content/en/integrations/guide/source-code-integration.md

```diff
@@ -540,7 +518,7 @@ If you're using the GitHub integration, click **Connect to preview** on error fr
 You can see links from failed test runs to their source repository in **Test Optimization**.
 
 1. Navigate to [**Software Delivery** > **Test Optimization** > **Test Runs**][101] and select a failed test run.
-2. Click the **View on GitHub** button to open the test in its source code repository.
+2. If you're using the GitHub or GitLab integrations, click **Connect to preview** on stack frames. You can see inline code snippets directly in the stack trace. Otherwise, you can always click the **View** button on the right of a frame or select **View file**, **View Git blame**, or **View commit** to be redirected to your source code management tool.
```

Comment:
```suggestion
2. If you're using the GitHub or GitLab integrations, click **Connect to preview** on stack frames. You can see inline code snippets directly in the stack trace. Otherwise, you can click the **View** button on the right of a frame or select **View file**, **View Git blame**, or **View commit** to be redirected to your source code management tool.
```